fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -653,17 +675,47 @@ void scanWithSynonymIndex() {
             // "hullo" is synonym of "hello"
             recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
             assertEquals(1, recordStore.scanIndex(SYNONYM_LUCENE_INDEX, IndexScanType.BY_LUCENE_FULL_TEXT,
-                            TupleRange.allOf(Tuple.from("text:(+\"hullo\" AND +\"record\" AND +\"layer\")")), null, ScanProperties.FORWARD_SCAN)
+                            TupleRange.allOf(Tuple.from(matchAll("hullo", "record", "layer"))), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
             // it doesn't match due to the "recor"
             recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
             assertEquals(0, recordStore.scanIndex(SYNONYM_LUCENE_INDEX, IndexScanType.BY_LUCENE_FULL_TEXT,
-                            TupleRange.allOf(Tuple.from("text:(+\"hullo\" AND +\"recor\" AND +\"layer\")")), null, ScanProperties.FORWARD_SCAN)
+                            TupleRange.allOf(Tuple.from(matchAll("hullo", "recor", "layer"))), null, ScanProperties.FORWARD_SCAN)
                     .getCount().join());
             // "hullo" is synonym of "hello", and "show" is synonym of "record"
             recordStore.saveRecord(createSimpleDocument(1623L, "Hello record layer", 1));
             assertEquals(1, recordStore.scanIndex(SYNONYM_LUCENE_INDEX, IndexScanType.BY_LUCENE_FULL_TEXT,
-                            TupleRange.allOf(Tuple.from("text:(+\"hullo\" AND +\"show\" AND +\"layer\")")), null, ScanProperties.FORWARD_SCAN)
+                            TupleRange.allOf(Tuple.from(matchAll("hullo", "show", "layer"))), null, ScanProperties.FORWARD_SCAN)
+                    .getCount().join());
+        }
+    }
+
+    @Test
+    void scanWithCombinedSetsSynonymIndex() throws IOException {
+        // The COMBINED_SYNONYM_SETS adds this extra line to our synonym set:
+        // 'synonym', 'nonsynonym'
+        try (InputStream is1 = SynonymAnalyzer.openDefaultFile();
+                InputStream is2 = SynonymAnalyzer.openFile("test.txt");
+                InputStream is = new SequenceInputStream(is1, is2)) {
+            SynonymAnalyzer analyzer = new SynonymAnalyzer(
+                    EnglishAnalyzer.ENGLISH_STOP_WORDS_SET,
+                    COMBINED_SYNONYM_SETS,
+                    new InputStreamReader(is, StandardCharsets.UTF_8)
+            );
+            SynonymAnalyzerRegistryImpl.instance().register(analyzer);
```

Comment:
It might be a good idea to add an `@AfterEach` teardown method that resets the registry to avoid this causing problems in other tests. In theory, if we ever parallelize test execution, we'll also need to worry more about thread safety of the registry itself, but maybe that's a task for another day.